app/javascript/mastodon/store/middlewares/errors.ts

```diff
@@ -1,9 +1,11 @@
-import { showAlertForError } from '../actions/alerts';
+import { Middleware } from 'redux';
+import { showAlertForError } from '../../actions/alerts';
+import { RootState } from '..';
 
 const defaultFailSuffix = 'FAIL';
 
-export default function errorsMiddleware() {
-  return ({ dispatch }) => next => action => {
+export const errorsMiddleware: Middleware<Record<string, never>, RootState> =
```

Comment:
Sorry about the typed redux noob question, but what does the `Record<string, never>` mean here?

Reply:
This was a suggested change by the [`ban-types` eslint rule](https://typescript-eslint.io/rules/ban-types/`), instead of `{}`:
> Don't use `{}` as a type. `{}` actually means "any non-nullish value".
> - If you want a type meaning "any object", you probably want `object` instead.
> - If you want a type meaning "any value", you probably want `unknown` instead.
> - If you want a type meaning "empty object", you probably want `Record<string, never>` instead.
> - If you really want a type meaning "any non-nullish value", you probably want `NonNullable<unknown>` inste

 `Record<KeyType, ValuesType>` in Typescript represent an object with the given keys and values, and can also be written as `{ [key: KeyType]: ValuesType }`, but record is often more legible when you need a simple object (the other syntax can support read-only values and other advanced features)

Reply:
Yes, the types for Redux are a bit complex…

I used the official documentation for this: https://redux.js.org/usage/usage-with-typescript#type-checking-middleware

This first parameter is if your middleware add things to `dispatch`'s return value, which is not the case in our middlewares.